### PR TITLE
Image - New Component

### LINKED ISSUE #32: [Component] - Image

### PR DESCRIPTION
**Pull request by artemleznikov**
#32 
Simple Image Component for images which have backend processing and accept query params width and quality (Imgix, Contentful)

Features:
- Lazy load, starts load on intersect
- Reproduces Medium functionality, has an initial small low quality blurred fallback image, which is replaced by real image once loaded.
- Uses srcSet and sizes to serve the best image based on width and dpi
- Accepts 3 sizes for mobile, tablet and desktop
- Generates a list of srcSet's based on incremental width
image

### CODE REVIEW

File: src/components/Image/utils.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import noop from 'no-op';
```
Comment (realdeemy) [inline src/components/Image/utils.js]: is there any way we could use https://github.com/Jam3/react-hooks/blob/master/src/hooks/use-intersection-observer.js instead?

File: src/components/Image/Image.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      if (!lazyLoad) return;
+
+      const observer = intersectionObserver(ImageElRef.current, {
+        onIntersectionIn: () => {
+          setLoadImage(true);
+        },
+        viewportTriggerPercentageTop: -5,
+        viewportTriggerPercentageBottom: -5,
+        viewportTriggerPercentageLeft: -5,
+        viewportTriggerPercentageRight: -5
```
Comment (realdeemy) [inline src/components/Image/Image.js]: should we expose these in props and have -5 as default?

File: src/components/Image/Image.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.Image {
+  display: block;
+  width: 100%;
+  margin-top: 4000px;
```
Comment (realdeemy) [inline src/components/Image/Image.scss]: curious about this value

**Review by realdeemy — COMMENTED** [review]
@artemleznikov nice.
I see it's a WIP but leaving some early comments

File: src/components/Image/Image.scss (hunk 3 above)
Comment (artemleznikov) [reply src/components/Image/Image.scss]: Just a temporary value to test intersection observer.

**Review by realdeemy — APPROVED** [review]
@artemleznikov lgtm 👍

please rebase update the minor things in the comments below and it's ready to go!

File: src/components/Image/Image.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
+import React, { memo, forwardRef, useRef, useEffect, useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import checkProps from '@jam3/react-check-extra-props';
+import { useIntersectionObserver } from '@jam3/react-hooks';
+
+import './Image.css';
```
Comment (realdeemy) [inline src/components/Image/Image.js]:
```suggestion
import './Image.scss';
```
should be supported now

File: src/components/Image/Image-story.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import Component from './Image';
+
+const props = {
+  src: 'https://assets.imgix.net/unsplash/mountains.jpg'
+};
+
+storiesOf('Image', module).addWithJSX('Default', () => <Component {...props} />);
```
Comment (realdeemy) [inline src/components/Image/Image-story.js]:
```suggestion
storiesOf('Image', module).add('Default', () => <Component {...props} />);
```
use `add` after rebase for the new storybook version

**Pull request merged**